src/gridstack-dd.ts

```diff
@@ -452,6 +453,11 @@ GridStack.prototype._prepareDragDropByNode = function(node: GridStackNode): Grid
       if (resizing && node.subGrid) { (node.subGrid as GridStack).onParentResize(); }
       this._updateContainerHeight();
     }
+    this._writePosAttr(target, node.x, node.y, node.w, node.h);

```

Comment:
this all needs to move inside the if (this.engine.moveNodeCheck(node, x, y, w, h)) {}
otherwise it didn't change (constrain).